### PR TITLE
Ensure support to torch 2.0

### PR DESCRIPTION
**Pull request by johnnv1**
Update the CI config to use torch `2.0.0` instead of `1.13.1`

### CODE REVIEW

File: .github/workflows/pr_test_typing.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ jobs:
       matrix:
         os: ['Ubuntu']
         python-version: ['3.7']
```
Comment (johnnv1) [inline .github/workflows/pr_test_typing.yml:21]:
```suggestion
        python-version: ['3.8']
```

File: kornia/augmentation/random_generator/base.py [hunk 2]
```diff
@@ -83,19 +83,19 @@ def __init__(self, dist: Distribution, map_fn: Optional[Callable[[Tensor], Tenso
...
     def sample(self, sample_shape: Tuple[int, ...]) -> Tensor:  # type: ignore[override]
-        out = self.dist.sample(sample_shape)
+        out = self.dist.sample(torch.Size(sample_shape))
         if self.map_fn is not None:
             out = self.map_fn(out)
         return out
 
     def sample_n(self, n: Union[int, float, Tensor]) -> Tensor:
-        out = self.dist.sample_n(n)
+        out = self.dist.sample_n(int(n.item()) if isinstance(n, Tensor) else int(n))
```
Comment (johnnv1) [inline kornia/augmentation/random_generator/base.py:98]: @shijianjian seems sample_n should accept just integer, should I do this casting or just update the type annotation? 

```output
kornia/augmentation/random_generator/base.py:98: error: Argument 1 to
"sample_n" of "Distribution" has incompatible type "Union[int, float, Tensor]";
expected "int"  [arg-type]
            out = self.dist.sample_n(n)
```

**Comment by johnnv1** [comment]
i think the failing test:
> test/augmentation/test_augmentation.py::TestRandomRGBShift::test_onnx_export[cpu-float32] - torch.onnx.errors.OnnxExporterError: Module onnx is not installed!

seems to be a problem on PyTorch side, but not sure, some idea here ? cc\ @edgarriba @ducha-aiki

**Comment by edgarriba** [comment]
Maybe they removed onnx and need to be installed by hand during tests ?

**Comment by johnnv1** [comment]
> Maybe they removed onnx and need to be installed by hand during tests ?

seems to be a bug with 2.0, and to be working on nightlies (2.1) -- will just skip the test for the 2.0

**Comment by johnnv1** [comment]
Failing cases:
- Ubuntu
  - Python 3.8
    - float 32
      - TestCanny.test_dynamo: Crazy crash on dynamo C++
        - [ ] This one I will skip for 2.0, already working on nightlies 
    - Float 64
      - TestWarpPerspective::test_dynamo: Tensor-likes are not close!
      - TestLovaszSoftmaxLoss::test_dynamo: Scalars are not close!
  - Python 3.10
    - Float 64
      - TestWarpPerspective::test_dynamo: Tensor-likes are not close!
      - TestLovaszSoftmaxLoss::test_dynamo: Scalars are not close!
- Windows
  - Python 3.8
    - float 32
      - TestImageRegistrator::test_registration_real[cpu-float32-loftr_homo] - AssertionError: Tensor-likes are not close!

File: test/color/test_gray.py [hunk 3]
```diff
@@ -201,7 +201,9 @@ def test_gradcheck(self, device, dtype):
         img = torch.ones(B, C, H, W, device=device, dtype=torch.float64, requires_grad=True)
         assert gradcheck(kornia.color.rgb_to_grayscale, (img,), raise_exception=True, fast_mode=True)
 
+    @pytest.mark.skip(reason='Is crashing on compile')
```
Comment (johnnv1) [inline test/color/test_gray.py:204]: Skip just for 2.0

File: test/color/test_gray.py [hunk 4]
```diff
@@ -292,7 +294,9 @@ def test_gradcheck(self, device, dtype):
         img = torch.ones(B, C, H, W, device=device, dtype=torch.float64, requires_grad=True)
         assert gradcheck(kornia.color.bgr_to_grayscale, (img,), raise_exception=True, fast_mode=True)
 
+    @pytest.mark.skip(reason='Is crashing on compile')
```
Comment (johnnv1) [inline test/color/test_gray.py:297]: Skip just for 2.0

File: test/augmentation/test_augmentation.py [hunk 5]
```diff
@@ -4036,6 +4037,7 @@ def test_smoke(self, device, dtype):
         out = aug(img)
         assert out.shape == (2, 3, 4, 5)
 
+    @pytest.mark.skipif(torch_version() == '2.0.0', reason='Not working on 2.0')
```
Comment (edgarriba) [inline test/augmentation/test_augmentation.py:4040]: Open later issues or directly to pytorch

**Review by edgarriba — APPROVED** [review]

File: test/augmentation/test_augmentation.py (hunk 5 above)
Comment (johnnv1) [reply test/augmentation/test_augmentation.py:4040]: It seems to be already fixed on the nightly version (2.1) 😄

Comment (edgarriba) [reply test/augmentation/test_augmentation.py:4040]: then why not supporting 2.1 directly ? in the past we had similar issues and we ended up supporting versions not ended with patch version == 0

Comment (johnnv1) [reply test/augmentation/test_augmentation.py:4040]: since they just release the 2.0, I did not expect the 2.1 to be out already this month

Comment (edgarriba) [reply test/augmentation/test_augmentation.py:4040]: ok, then let's merge this as it is

**Pull request merged**